sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -190,22 +191,45 @@ struct TransportException : public std::exception
   const char* what() const throw() { return "Error on transport layer while sending request"; }
 };
 
+class HttpVersion
+{
+private:
+  uint16_t m_mayorVersion;
+  uint16_t m_minorVersion;
+  static const std::string m_title;
+  static const std::string m_separator;
+
+public:
+  HttpVersion(uint16_t mayorVersion, uint16_t minorVersion)
+      : m_mayorVersion(mayorVersion), m_minorVersion(minorVersion)
+  {
+  }
+
+  HttpVersion() : HttpVersion(0, 0) {}
```

Comment:
What is operator:string for? Logging? Some comments would be nice.
Do we really even need this class?
Is it legal to create an HttpVersion with 0,0? What would this mean?

Reply:
so, maybe the constructor with (0,0) is not needed. I just didn´t know if by default it would add garbage to fields, so I explicitly set it to 0.

operator string is use similar to .toString() method from C#/Java, it means we can use the instance directly as std::string like `cout << httpVersion;` 

I am currently expecting this class not to be visible to customers and only to be useful internally to organize code a little better.
I though that any validation we might want to have for checking version or for any other functionality  would be isolated from the Response itself and just as a httpVersion thing.
In C, for instance, we created an struct to hold minor and mayor version.

But, I am not against having this along with Response.  Let me know if after my thinking you would still prefer to have them merged and I will merge with Response Class